Update the contributing to docs guide (#2061)
o3de/o3de.org: content/docs/contributing/to-docs/get-started.md

---
linktitle: Get Started
title: Get Started Contributing to Docs
description: Get started contributing to Open 3D Engine (O3DE) docs by setting up your local repo, setting up your writing environment, and submitting docs. 
toc: true
weight: 200
---

This section will help you get started on contributing to Open 3D Engine (O3DE) documentation. You'll learn how to set up your writing environment and about the processes you need to get your documentation from text editor to the official O3DE documentation. 

O3DE documentation uses the [fork and pull model](https://en.wikipedia.org/wiki/Fork_and_pull_model) for contributions. As a contributor, you maintain a fork (your own repository of the O3DE documentation) on GitHub, and work locally to edit documentation. You then submit PRs from your fork for review.

{{< note >}}

> The steps on this page use the terminal to run Git commands. You may complete these steps through an alternative tool of your choice, but ensure that you're completing the correct operations. 

Reviewer: nit: Alternative is most likely the GitHub UX, do you want to link to their instructions?

Author: Adding information on this

{{< /note >}}

## Prerequisites

* Sign up for a GitHub account here [Join GitHub](https://github.com/join?ref_cta=Sign+up).

* Install **Git** version control software. Get Git here [Git Downloads](https://git-scm.com/downloads).

* Install an editor for making changes to Markdown (`.md`) files. You can use any editor, but we do recommend one that supports Markdown linting. VS Code is commonly used by contributors and you can get it here [Microsoft VS Code](https://code.visualstudio.com/download).

## Setting up a local o3de.org repo

In this section, you'll learn to create your own fork of the O3DE documentation. 

### Create a fork

> A *fork* is your own copy of `o3de.org:main` on GitHub. You can do anything you like within your fork, though it is recommended that you keep your fork synced with `o3de.org:main`, and work within branches in your fork. Working this way ensures the integrity of `o3de.org:main` and makes it easy for you to work at your own pace, experiment with changes, and collaborate with other contributors. 

Reviewer: Do we need to cover the difference between o3de.org:main and o3de.org:development here?

Author: Yep! Creating a section to distinguish the upstream branches. Also amending this section, the fork operation occurs on the whole repo, and i don't believe is dependent on the branch.

{{< /important >}}

### Preview your docs

As you write, you should preview of your docs to ensure it's in the correct format. Because the o3de.org website is built on Hugo, the best way to preview your docs is by running Hugo on your local machine. For information on how to set up and use Hugo, refer to the o3de.org [`README.md`](https://github.com/o3de/o3de.org) file. 

Additionally, a quick way to preview your docs is by using VS Code's [Editor and preview synchronization](https://code.visualstudio.com/docs/languages/markdown#_editor-and-preview-synchronization) feature for Markdown. This feature is limited to VS Code's Markdown support, so it won't render any features that are unique to Hugo, such as shortcodes. 

Finally, you can preview and share a preview of your docs when they are in a PR. Netlify deploys a preview for each PR. This takes a few minutes to deploy after creating the PR, and refreshes everytime you push a commit. You can access the deployed preview at the bottom of the PR webpage: Find **netlify/o3deorg/deploy-preview — Deploy Preview ready!** and click **Details**. 

## Submitting docs

> When you make changes or create new docs, you must submit a pull request (PR) for review before the changes can be merged into `o3de.org:main`. PRs allow peer contributors to review contributions for several potential issues, including technical accuracy, spelling, grammar, clarity, and style. 

Reviewer: Minor: I would argue to create a pull request, submit is the action to merge it.

* To commit multiple suggestions, use the batch functionality to integrate multiple suggestions in a single commit.
* Do try to acknowledge all the feedback you are given. This doesn't mean to integrate every comment and commit every suggestion. It simply means to avoid resolving comments and suggestions without action or a response. Embrace collaboration.
* When addressing comments, maintain the relative conversation in the PR, edit the topics as necessary, and commit the changes to your fork/branch. Your new commits will be automatically added to the PR.
* Make sure to request a re-review of your new commits if required. More guidance on how to traverse our PR process is coming!

For more information on incorporating PR feedback, refer to [Incorporating feedback in your pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request).

### The PR merge

> When you have approval from two reviewers, and approval from a technical reviewer (if necessary), your PR can be merged. Never merge your own PRs. The last reviewer to approve the PR is responsible for merging the PR.

Reviewer: I believe only Docs maintainers should merge PRs.

Author: Yes was understood initially, but not always practiced. Especially since the `write` access was changed to include reviewers from other domains. I'll update this text.

A DCO signature appears in the last part of your commit message in the form `Signed-off-by: user.name <user.email>"`, where _user.name_ and _user.email_ are the `user.name` and `user.email`, respectively, from your `.gitconfig` file.

{{< important >}}
In some cases, you may need to manually add a `Signed-off-by:` line to your commit message, for example if you're using a GUI tool that doesn't support DCO sign off. 
For GitHub accounts created prior to July 18, 2017, use `username@users.noreply.github.com` as the email address. For accounts created after that date, use your GitHub-provided no-reply email address. The new no-reply email address is a seven-digit ID number and your username in the form of `ID+username@users.noreply.github.com` which can be found in the email tab of your GitHub account settings. For more information about setting your commit email address, refer to the [Setting your commit email address instructions](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address).
{{< /important >}}

### Preview your docs

> As you write, you should preview of your docs to ensure it's in the correct format. Because the o3de.org website is built on Hugo, the best way to preview your docs is by running Hugo on your local machine. For information on how to set up and use Hugo, refer to the o3de.org [`README.md`](https://github.com/o3de/o3de.org) file. 

Reviewer 2: Improve link:
```suggestion
As you write, you should preview of your docs to ensure it's in the correct format. Because the o3de.org website is built on Hugo, the best way to preview your docs is by running Hugo on your local machine. For information on how to set up and use Hugo, refer to the o3de.org [`README.md`](https://github.com/o3de/o3de.org#readme) file. 
```

Additionally, a quick way to preview your docs is by using VS Code's [Editor and preview synchronization](https://code.visualstudio.com/docs/languages/markdown#_editor-and-preview-synchronization) feature for Markdown. This feature is limited to VS Code's Markdown support, so it won't render any features that are unique to Hugo, such as shortcodes. 

> Finally, you can preview and share a preview of your docs when they are in a PR. Netlify deploys a preview for each PR. This takes a few minutes to deploy after creating the PR, and refreshes everytime you push a commit. You can access the deployed preview at the bottom of the PR webpage: Find **netlify/o3deorg/deploy-preview — Deploy Preview ready!** and click **Details**. 

Reviewer 2: Recommend that you give a hint what Netlify is, since many in the target audience will be unfamiliar.
```suggestion
Finally, you can view and share a preview of your docs when they are in a PR. The Netlify web deployment service creates a preview for each PR. This takes a few minutes to deploy after creating the PR, and refreshes every time you push a commit. You can access the deployed preview at the bottom of the PR webpage: Find **netlify/o3deorg/deploy-preview — Deploy Preview ready!** and click **Details**. 
```

## Submitting docs

> When you make changes or create new docs, you must create a pull request (PR) for review before the changes can be merged into `o3de.org:main`. PRs allow peer contributors to review contributions for several potential issues, including technical accuracy, spelling, grammar, clarity, and style. 

Reviewer 2: ```suggestion
When you make changes or create new docs, you must create a pull request (PR) for review before the changes can be merged into `o3de.org`. PRs allow peer contributors to review contributions for several potential issues, including technical accuracy, spelling, grammar, clarity, and style. 
```

The current list of active PRs is here [O3DE repository pull requests (PRs)](https://github.com/o3de/o3de.org/pulls).

### Create a PR

PRs are created in the GitHub web interface from your branch. Go to your fork on GitHub and perform the steps below.

1. In the **Pull Request** tab, select the green **New pull request** button to create a new PR.

1. In the **Comparing changes** page, you will compare your fork's branch against the remote source's branch. Ensure that the **base**, which you're comparing against, points to `o3de/o3de.org:main` or `o3de/o3de.org:development`. The **head repository** should be your fork. Then, in the **compare** dropdown, choose the branch that you want to submit.

> 1. Verify that the changed files contains only the changes you want to submit. Only the commits you made should be listed. If there are more commits than you expected, stop here and make sure to properly sync your branch.

Reviewer 2: There are several sections describing how to sync in this topic. A link to these specific iteration steps that you introduced earlier would help clarify here:
```suggestion
1. Verify that the changed files contains only the changes you want to submit. Only the commits you made should be listed. If there are more commits than you expected, stop here and make sure to properly sync your branch using the fetch, merge, and push commands from the [iteration steps](#sync-write-add-commit-and-push) explained earlier.
```

### Respond to feedback

Feedback comes in the form of comments that you can address by editing files in your local branch, and *suggestions* that can be committed from the GitHub PR interface. It's important to understand that the PR process is a collaborative discussion. Every comment does not need to be addressed, and every suggestion does not need to be integrated. When the required number of reviewers approve of your contribution, it can be integrated. Here are a few tips for addressing suggestions and comments:

* To commit multiple suggestions, use the batch functionality to integrate multiple suggestions in a single commit.
* Do try to acknowledge all the feedback you are given. This doesn't mean to integrate every comment and commit every suggestion. It simply means to avoid resolving comments and suggestions without action or a response. Embrace collaboration.
* When addressing comments, maintain the relative conversation in the PR, edit the topics as necessary, and commit the changes to your fork/branch. Your new commits will be automatically added to the PR.

> * Make sure to request a re-review of your new commits if required. More guidance on how to traverse our PR process is coming!

Reviewer 2: ```suggestion
* Make sure to request a re-review of your new commits if required.
```

* To commit multiple suggestions, use the batch functionality to integrate multiple suggestions in a single commit.
* Do try to acknowledge all the feedback you are given. This doesn't mean to integrate every comment and commit every suggestion. It simply means to avoid resolving comments and suggestions without action or a response. Embrace collaboration.
* When addressing comments, maintain the relative conversation in the PR, edit the topics as necessary, and commit the changes to your fork/branch. Your new commits will be automatically added to the PR.
* Make sure to request a re-review of your new commits if required.

For more information on incorporating PR feedback, refer to [Incorporating feedback in your pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request).

### The PR merge

When you have an approval from at least one docs reviewer and at least one technical reviewer, your PR can be merged. Never merge your own PRs. Docs maintainers are responsible for merging the PR.

> =======

Reviewer 2: Uh oh. Looks like the content from the old "Get Started" got merged in here in the [last commit](https://github.com/o3de/o3de.org/pull/2061/commits/2262c078f68d3a54d9e77859c3dd03cee3092576). I think you can delete everything from this metadata section to the end, but I'm not sure if there's anything in here that was supposed to go into the new overview.md?
